### DISCUSSION TITLE
How does the circuit of DonConIO work？

### DISCUSSION THREAD

**Discussion by HellbroWDR** [ravinrabbid/DonCon2040#3] [Q&A] [answered]
I read your pdf of DonConIO circuit and look up some data on google, but I can’t really know how the circuit works. Can someone explain it?
I think it is combination of transport amplifier and peak detector, but I don’t know yes or not.

**Comment by ravinrabbid** [comment, marked as answer]
Take this with a grain of salt since I'm not an electrical engineer:

- First a RC high-pass filter to get rid of the vibrations that traveled through the rubber buffers from other pads
- An op-amp with high impedance to avoid putting load on the (very low power) signal, also amplifying the signal by factor 20
- A diode for half-wave rectification followed by a peak detector
- A second op-amp buffer to avoid loading the peak detector
- Some diodes to protect the circuit as well as the inputs of the µC/ADC

**Reply by HellbroWDR** [reply]
Thanks for you reply. I also want to ask what the voltage of Ka_Left_ADC is. I want to use this circuit but I want to use Arduino to control. However, the maximum voltage of analog input in Arduino is 5V.
In addition, I think you use MCP3204 to change analog to digital because rp2040 can only receive digital input. Is it right？

**Reply by ravinrabbid** [reply]
The voltage is capped to ~4.3V by the zener diode at the output.

The RP2040, or more specific the XIAO RP2040 board I used has analog inputs, but those are somewhat noisy and have a static offset. The MCP3204 is for better sensitivity. Depending on your construction the internal ADCs of the Arduino might be sufficient. Using the Arduino with an external reference voltage could also be an option.
